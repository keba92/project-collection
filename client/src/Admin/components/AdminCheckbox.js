import React from 'react';

const AdminCheckbox = ({ type = "checkbox", checked = false, onChange, title, value }) => {
    return (
      <input type={type} className ='form-check-input' checked={checked} onChange={onChange} title ={title} value={value} />
    );
};

export default AdminCheckbox;