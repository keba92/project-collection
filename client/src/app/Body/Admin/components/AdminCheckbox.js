import React, { memo } from 'react';
import PropTypes from 'prop-types';

function AdminCheckbox(props) {
  const { type, checked, onChange, title, value } = props;
  return (
    <input
      type={type}
      className="form-check-input"
      checked={checked}
      onChange={onChange}
      title={title}
      value={value}
    />
  );
}

AdminCheckbox.defaultProps = {
  type: 'checkbox',
  checked: false,
};

AdminCheckbox.propTypes = {
  type: PropTypes.string,
  checked: PropTypes.bool,
  onChange: PropTypes.func,
  title: PropTypes.string,
  value: PropTypes.string,
};

export default memo(AdminCheckbox);
