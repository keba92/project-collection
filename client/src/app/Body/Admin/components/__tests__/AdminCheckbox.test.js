import React from 'react';
import AdminCheckbox from '../AdminCheckbox';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

afterEach(cleanup);

const testFn = ()=>{
    return 'test';
}

it('matches snapshot', () => {
    const tree = renderer.create(<AdminCheckbox
                                    checked='user'
                                    onChange={testFn}
                                    value='user_id'
                                />).toJSON();
    expect(tree).toMatchSnapshot();
})