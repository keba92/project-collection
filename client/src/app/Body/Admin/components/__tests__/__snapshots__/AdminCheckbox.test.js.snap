// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<input
  checked="user"
  className="form-check-input"
  onChange={[Function]}
  type="checkbox"
  value="user_id"
/>
`;
